frontend/src/metabase/dashboard/components/ClickBehaviorSidebar/LinkOptions/LinkedEntityPicker/LinkedEntityPicker.tsx

```diff
@@ -97,32 +101,57 @@ function getTargetClickMappingsHeading(entity: Question | Dashboard) {
 }
 
 function TargetClickMappings({
-  isDashboard,
   clickBehavior,
   dashcard,
   updateSettings,
 }: {
-  isDashboard: boolean;
   clickBehavior: EntityCustomDestinationClickBehavior;
   dashcard: DashboardCard;
   updateSettings: (settings: Partial<ClickBehavior>) => void;
 }) {
-  const Entity = isDashboard ? Dashboards : Questions;
+  const isDashboard = clickBehavior.linkType === "dashboard";
```

Comment:
If we pass `isDashboard` as a prop, we lose typing information. Computing it here allows TS to differentiate whether `clickBehavior` is `DashboardCustomDestinationClickBehavior` or `QuestionCustomDestinationClickBehavior`.